src/serializer/ResidualHeapSerializer.js

```diff
@@ -1101,6 +1101,13 @@ export class ResidualHeapSerializer {
                   mightHaveBeenDeleted
                 );
               });
+
+              // Although the property needs to be delayed, we still want to emit "undefined"
+              // as part of the object literal to ensure a consistent property ordering.
+              if (!mightHaveBeenDeleted) {
```

Comment:
Your change certainly improved things into the right direction, and checking for `!mightHaveBeenDeleted` makes sense.

Even better, but that could come in a different pull request, would be if you always emit the `undefined` entry here, but then issue a (conditional) `delete` statement afterwards if the entry got deleted. In `_assignProperty`, the assignment currently happens conditionally, and there an `else` branch could be generated that issue the deletion, assuming that your new kicks in that adds a dummy value to begin with (`_assignProperty` is called from a few more places.)